miio/cooker_multi.py

```diff
@@ -0,0 +1,419 @@
+            if schedule is not None and schedule > 0 and schedule <= 1440:
+                self.set_schedule_enabled(True)
+                self.set_schedule_duration(schedule)
+            if akw is not None:
+                self.set_akw_enabled(akw)
+
+    def is_set_duration_allowed(self):
+        return (
+            self.profile_bytes[10] != self.profile_bytes[12]
+            or self.profile_bytes[11] != self.profile_bytes[13]
+        )
+
+    def get_duration(self):
+        """Get the duration in minutes."""
+        return (self.profile_bytes[8] * 60) + self.profile_bytes[9]
+
+    def set_duration(self, minutes):
+        """Set the duration in minutes if the profile allows it."""
+        if not self.is_set_duration_allowed():
+            return
+
+        max_minutes = (self.profile_bytes[10] * 60) + self.profile_bytes[11]
+        min_minutes = (self.profile_bytes[12] * 60) + self.profile_bytes[13]
+
+        if minutes < min_minutes or minutes > max_minutes:
+            return
+
+        self.profile_bytes[8] = math.floor(minutes / 60)
+        self.profile_bytes[9] = minutes % 60
+
+        self.update_checksum()
+
+    def is_schedule_enabled(self):
+        return (self.profile_bytes[14] & 0x80) == 0x80
+
+    def set_schedule_enabled(self, enabled):
+        if enabled:
+            self.profile_bytes[14] |= 0x80
+        else:
+            self.profile_bytes[14] &= 0x7F
+
+        self.update_checksum()
+
+    def set_schedule_duration(self, duration):
+        """Set the schedule time (delay before cooking) in minutes."""
+        schedule_flag = self.profile_bytes[14] & 0x80
+        self.profile_bytes[14] = math.floor(duration / 60) & 0xFF
+        self.profile_bytes[14] |= schedule_flag
+        self.profile_bytes[15] = (duration % 60 | self.profile_bytes[15] & 0x80) & 0xFF
+
+        self.update_checksum()
+
+    def is_akw_enabled(self):
+        return (self.profile_bytes[15] & 0x80) == 0x80
+
+    def set_akw_enabled(self, enabled):
+        if enabled:
+            self.profile_bytes[15] |= 0x80
+        else:
+            self.profile_bytes[15] &= 0x7F
+
+        self.update_checksum()
+
+    def calc_checksum(self):
+        crc = crcmod.mkCrcFun(0x11021, rev=False, initCrc=0x0, xorOut=0x0)(
+            self.profile_bytes
+        )
+        checksum = bytearray(2)
+        checksum[0] = (crc >> 8) & 0xFF
+        checksum[1] = crc & 0xFF
+        return checksum
+
+    def update_checksum(self):
+        self.checksum = self.calc_checksum()
+
+    def is_valid(self):
+        return len(self.profile_bytes) == 174 and self.checksum == self.calc_checksum()
+
+    def get_profile_hex(self):
+        return (self.profile_bytes + self.checksum).hex()
+
+
+class CookerStatus(DeviceStatus):
+    def __init__(self, data):
+        self.data = data
+
+    @property
+    def mode(self) -> OperationMode:
+        """Current operation mode."""
+        return OperationMode(self.data["status"])
```

Comment:
Please add the decorators for entries that should be exposed per default to cli and homeassistant, see https://python-miio.readthedocs.io/en/latest/contributing.html#status-containers or how it's done in yeelight or roborock integrations.

Basically, everything that is properly decorated will be automatically printed out when calling status() and are also exposed to the [next-gen xiaomi_miio integration](https://github.com/rytilahti/homeassistant-xiaomi-ng) whenever it gets ready.

Reply:
done